In [14]:
print("Hello Udhay")

Hello Udhay


# Statistical Analysis for Decision Making
## HealthCare Plus Hospital

### Question 1

Compute the Mean, Median, and Mode of patient admissions.

Admissions Data:
[32, 28, 35, 30, 29, 27, 31, 34, 33, 30]

In [15]:
import numpy as np
from scipy import stats

admissions = [32, 28, 35, 30, 29, 27, 31, 34, 33, 30]

mean = np.mean(admissions)
median = np.median(admissions)
mode = stats.mode(admissions)

print("Mean =", mean)
print("Median =", median)
print("Mode =", mode.mode)

Mean = 30.9
Median = 30.5
Mode = 30


### Interpretation

- Mean = 30.9
- Median = 30.5
- Mode = 30

The mean best represents patient admissions because there are no extreme values in the dataset.

If admission capacity increases by 10%, the mean, median, and mode will also increase by approximately 10%.

# Question 2

Calculate the Range, Variance, and Standard Deviation of recovery durations.

Recovery Data:
[5, 7, 6, 8, 9, 5, 6, 7, 8, 6]

In [16]:
import numpy as np

recovery = [5, 7, 6, 8, 9, 5, 6, 7, 8, 6]

range_value = max(recovery) - min(recovery)
variance = np.var(recovery)
std_dev = np.std(recovery)

print("Range =", range_value)
print("Variance =", variance)
print("Standard Deviation =", std_dev)

Range = 4
Variance = 1.61
Standard Deviation = 1.2688577540449522


### Interpretation

- Range = 4
- Variance = 1.61
- Standard Deviation = 1.27

The standard deviation is low, indicating that recovery times are close to the average.

If two new patients recover in 4 and 10 days, the standard deviation will increase because the data becomes more spread out.

# Question 3

Compute Skewness and Kurtosis of patient satisfaction scores.

Satisfaction Scores:
[8, 9, 7, 8, 10, 7, 9, 6, 10, 8, 7, 9]

In [17]:
from scipy.stats import skew, kurtosis

satisfaction = [8, 9, 7, 8, 10, 7, 9, 6, 10, 8, 7, 9]

print("Skewness =", skew(satisfaction))
print("Kurtosis =", kurtosis(satisfaction))

Skewness = -0.04146734005998014
Kurtosis = -1.0145959416162336


### Interpretation

- Skewness = -0.041
- Kurtosis = -1.015

The skewness value is very close to zero, indicating that the data is approximately symmetric.

The negative kurtosis value indicates that the distribution is flatter than a normal distribution and contains fewer extreme values.

The satisfaction scores are approximately normally distributed.

If customer service improves and satisfaction scores increase, the distribution may become negatively skewed because more patients will give higher ratings.

# Question 4

Compute the correlation coefficient between nurse staffing and patient recovery time.

In [18]:
import numpy as np

nurses = [10, 12, 15, 18, 20, 22]
recovery_time = [8, 7, 6, 5, 4, 3]

correlation = np.corrcoef(nurses, recovery_time)[0,1]

print("Correlation =", correlation)

Correlation = -0.9967569377294726


### Interpretation

- Correlation = -0.9968

There is a very strong negative correlation between nurse staffing and recovery time.

As the number of nurses increases, patient recovery time decreases.

If the hospital increases the number of nurses by 5 per department, recovery time is expected to decrease further.

# Question 5

Test whether the hospital's claim that the average patient wait time is 30 minutes is valid.

In [19]:
from scipy import stats

wait_time = [32, 29, 31, 34, 33, 27, 30, 28, 35, 26]

t_stat, p_value = stats.ttest_1samp(wait_time, 30)

print("T-Statistic =", t_stat)
print("P-Value =", p_value)

T-Statistic = 0.5222329678670935
P-Value = 0.6141172548083939


### Interpretation

Null Hypothesis (H₀):
The average patient wait time is 30 minutes.

Alternative Hypothesis (H₁):
The average patient wait time is not 30 minutes.

P-Value = 0.6141

Since the p-value is greater than 0.05, we fail to reject the null hypothesis.

Therefore, the hospital's claim that the average wait time is 30 minutes is valid.

If waiting times increase, the hospital should increase staffing, improve scheduling, and use digital queue management systems.

# Question 6

Perform a Chi-Square Test to determine whether hospital cleanliness and patient satisfaction are dependent.

In [20]:
from scipy.stats import chi2_contingency

data = [
    [90, 10],
    [60, 40],
    [30, 70]
]

chi2, p, dof, expected = chi2_contingency(data)

print("Chi-Square =", chi2)
print("P-Value =", p)

Chi-Square = 75.0
P-Value = 5.175555005801876e-17


### Interpretation

Null Hypothesis (H₀):
Hospital cleanliness and patient satisfaction are independent.

Alternative Hypothesis (H₁):
Hospital cleanliness and patient satisfaction are dependent.

Chi-Square = 75.0
P-Value = 5.17 × 10⁻¹⁷

Since the p-value is less than 0.05, we reject the null hypothesis.

There is a significant relationship between hospital cleanliness and patient satisfaction.

Improved cleanliness will increase the number of satisfied patients and decrease the number of unsatisfied patients.

# Question 7

Conduct an ANOVA test to determine whether there is a significant difference in recovery times among Treatment A, B, and C.

In [21]:
from scipy.stats import f_oneway

A = [5, 6, 7, 5, 6]
B = [8, 9, 7, 8, 10]
C = [4, 5, 6, 5, 4]

f_stat, p_value = f_oneway(A, B, C)

print("F-Statistic =", f_stat)
print("P-Value =", p_value)

F-Statistic = 19.18518518518518
P-Value = 0.0001828254331459655


### Interpretation

Null Hypothesis (H₀):
There is no significant difference in recovery times among Treatments A, B, and C.

Alternative Hypothesis (H₁):
There is a significant difference in recovery times among Treatments A, B, and C.

F-Statistic = 19.185
P-Value = 0.0001828

Since the p-value is less than 0.05, we reject the null hypothesis.

There is a significant difference in recovery times among the treatment methods.

Before introducing a new Treatment D, the hospital should collect recovery duration data, patient satisfaction scores, side effects, and treatment success rates.

# Question 8

Analyze whether administration times follow a normal distribution.

In [22]:
import numpy as np
from scipy import stats

admin_time = [12,15,14,16,18,13,14,17,15,19,16,14]

print("Mean =", np.mean(admin_time))
print("Standard Deviation =", np.std(admin_time))

shapiro_test = stats.shapiro(admin_time)

print("Shapiro Statistic =", shapiro_test.statistic)
print("P-Value =", shapiro_test.pvalue)

Mean = 15.25
Standard Deviation = 1.963203164898291
Shapiro Statistic = 0.970912163004773
P-Value = 0.9201214036707512


### Interpretation

Mean = 15.25
Standard Deviation = 1.96
P-Value = 0.9201

Since the p-value is greater than 0.05, the data follows a normal distribution.

This analysis is important because normality assumptions are required for many statistical methods used in healthcare.

If emergency cases increase, administration times may become positively skewed due to longer processing times.

# Question 9

Model emergency patient arrivals using a probability distribution.

In [23]:
from scipy.stats import poisson

probability = poisson.pmf(3, 5)

print("Probability =", probability)

Probability = 0.1403738958142805


### Interpretation

The appropriate probability distribution is the Poisson Distribution.

Probability of exactly 3 emergency arrivals:

P(X = 3) = 0.1404

There is approximately a 14.04% chance that exactly 3 emergency cases will arrive in the next hour.

If a major accident occurs, the average arrival rate will increase and the distribution will shift toward higher arrival counts.

# Question 10

Calculate the expected number of surgeries performed per day.

In [24]:
surgeries = [0, 1, 2, 3, 4, 5]
frequency = [5, 12, 18, 22, 15, 8]

total_frequency = sum(frequency)

expected_value = sum(x * f for x, f in zip(surgeries, frequency)) / total_frequency

print("Expected Surgeries Per Day =", expected_value)

Expected Surgeries Per Day = 2.675


### Interpretation

Expected Surgeries Per Day = 2.675

The hospital performs approximately 2.68 surgeries per day on average.

If a new surgical team is hired, the probability distribution may shift toward higher numbers of surgeries because the hospital will have increased capacity.

# Conclusion

1. Average patient admissions are approximately 31 per day.
2. Recovery times show low variability.
3. Patient satisfaction scores are approximately normally distributed.
4. Nurse staffing strongly affects recovery time.
5. The hospital's claim of a 30-minute average wait time is valid.
6. Hospital cleanliness significantly affects patient satisfaction.
7. Treatment methods differ significantly in effectiveness.
8. Administration times follow a normal distribution.
9. Emergency arrivals follow a Poisson distribution.
10. The hospital performs approximately 2.68 surgeries per day.

Statistical analysis helps HealthCare Plus make data-driven decisions to improve patient care, optimize hospital operations, and enhance overall efficiency.